lib/simplifyExpression/basicsSearch/convertMixedNumberToImproperFraction.js

```diff
@@ -0,0 +1,138 @@
+const clone = require('../../util/clone');
+const mixedNumber = require('./mixedNumber');
+
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+
+// Converts a mixed number to an improper fraction
+// e.g. 1 2/3 -> 5/3
+function convertMixedNumberToImproperFraction(node) {
+  if (!mixedNumber.isMixedNumber(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  const substeps = [];
+  let newNode = clone(node);
+
+  // e.g. 1 2/3
+  const wholeNumber = mixedNumber.getWholeNumber(node); // 1
+  const numerator = mixedNumber.getNumerator(node); // 2
+  const denominator = mixedNumber.getDenominator(node); // 3
+  const isNegativeMixedNumber = mixedNumber.isNegativeMixedNumber(node);
+
+  // STEP 1: Convert to unsimplified improper fraction
+  // e.g. 1 2/3 -> ((1 * 3) + 2) / 3
+  let status = convertToUnsimplifiedImproperFraction(
+    newNode, wholeNumber, numerator, denominator, isNegativeMixedNumber);
+  substeps.push(status);
+  newNode = clone(status.newNode);
+
+  // STEP 2: Simplify multiplication in numerator
+  // e.g. ((1 * 3) + 2) / 3 -> (3 + 2) / 3
+  status = simplifyMultiplicationInImproperFraction(
+    newNode, wholeNumber, numerator, denominator, isNegativeMixedNumber);
+  substeps.push(status);
+  newNode = clone(status.newNode);
+
+  // STEP 3: Simplify addition in numerator
+  // e.g. (3 + 2) / 3 -> 5/3
+  status = simplifyAdditionInImproperFraction(
+    newNode, wholeNumber, numerator, denominator, isNegativeMixedNumber);
+  substeps.push(status);
+  newNode = clone(status.newNode);
+
+  return Node.Status.nodeChanged(
+    ChangeTypes.CONVERT_MIXED_NUMBER_TO_IMPROPER_FRACTION,
+    node, newNode, true, substeps);
+}
+
+// Convert a mixed number to an unsimplified proper fraction
+// e.g. 1 2/3 -> ((1 * 3) + 2) / 3
+function convertToUnsimplifiedImproperFraction(
+  node, wholeNumber, numerator, denominator, isNegativeMixedNumber) {
+  let newNode = clone(node);
+
+  // (wholeNumber * denominator)
+  // e.g. (1 * 3)
+  const newNumeratorMultiplication = Node.Creator.parenthesis(
+    Node.Creator.operator(
+      '*', [Node.Creator.constant(wholeNumber),
+        Node.Creator.constant(denominator)]));
+
+  // (wholeNumber * denominator) + numerator
+  // e.g. (1 * 3) + 2
+  const newNumerator = Node.Creator.operator(
+    '+', [newNumeratorMultiplication,
+      Node.Creator.constant(numerator)]);
+  newNumerator.changeGroup = 1;
```

Comment:
if you set a changeGroup on the new node, it's often nice to set it on the old node too (I know Socratic doesn't use that for colours (I think?) but it's good practice to highlight in the tree both the before and after of a change)